src/libraries/System.Text.Json/gen/TypeGenerationSpec.cs

```diff
@@ -72,6 +72,11 @@ internal class TypeGenerationSpec
         public bool HasPropertyFactoryConverters { get; private set; }
         public bool HasTypeFactoryConverter { get; private set; }
 
+        // The spec is derived from cached `System.Type` instances, which are generally annotation-agnostic.
+        // Hence we can only record the potential for nullable annotations being possible for the runtime type.
+        // TODO: consider deriving the generation spec from the Roslyn symbols directly.
```

Comment:
Can we create a tracking issue for this and link it here?

Reply:
Yes, needed to satisfy the `ITestContext` implementation.